### DISCUSSION TITLE
File and path management

### DISCUSSION THREAD

**Discussion by deltamarnix** [pyphoenix/pyphoenix-project#53] [Ideas]
# Problem statement 

Any component, many blocks, variables, and even parts of variables (e.g. array layers) can be associated with input files (see MF6IO appendix for more details on open/close)

# Implementation

Looking into `MFFileMgmt`. There are only a couple of functions in this class that are supposed to be public.

- `copy_files()`. Only works when `last_loaded_sim_path` is not `None`.
- `get_model_path()`: Joins sim path with relative model path. If `last_loaded_path`, it will return those paths.
- `get_sim_path()`: If `last_loaded_path`, it will return that path.

The rest of the functions are for internal use only. Most functions are likely to be used in a static manner, but this last loaded path seems to make everything so much more complicated. I am wondering if we could redo the functionality and leave out anything that has to do with last loaded paths. Just give a destination if you want to copy something, for example.

**Comment by wpbonelli** [comment]
A few related considerations:

- should a simulation have a workspace property or should workspace be a parameter to `write_...()` methods?
- why have e.g. `copy_files()` when the user can do that outside of flopy?
- existing workspace approach vs separate input/output directories, cf https://github.com/modflowpy/flopy/issues/2462?
- can we disambiguate "binary" vs "external"? These terms seem to have some (maybe a lot of) overlap in flopy3
